polaris-core/src/main/java/org/apache/polaris/core/PolarisConfiguration.java

```diff
@@ -154,6 +154,14 @@ public static <T> Builder<T> builder() {
           .defaultValue(false)
           .build();
 
+  public static final PolarisConfiguration<Boolean> ALLOW_EXTERNAL_CATALOG_CREDENTIAL_VENDING =
```

Comment:
Since this is a catalog-level property, is there a need to make it specific to external catalogs? Admins could just choose to set this on only external catalogs.

Reply:
Admins could do that by just setting the flag on either INTERNAL or EXTERNAL catalogs, couldn't they? I don't see a reason to make the parameter specific to one type or the other.

Reply:
Looks like the current code will support different keys for yaml-level system-wide settings, while using the same catalogConfig for per-catalog overrides, which does seem to achieve both goals.

I would agree that the yaml-level config is the main thing that needs to be configurable separately by whoever is running the Polaris service.

If we can verify that it won't cause any problems to have the same `catalogConfig("enable.credential.vending")` for both `ALLOW_EXTERNAL_CATALOG_CREDENTIAL_VENDING` and `ALLOW_INTERNAL_CATALOG_CREDENTIAL_VENDING` then this seems like the best approach.

Reply:
I would agree with @eric-maynard and @sfc-gh-dhuo . Ideally I would probably like to be able to set this on a per catalog basis but having blanket coverage is nice too. In the future I would like to have a more flexible way defining policies like this.

Reply:
I'd also prefer a catalog property for this at this moment. We could consider policy-based one if there are solid use cases in the future.

Reply:
The current impl allows setting the default value for the service and allowing for catalog-level overrides. This is confirmed in the tests [here](https://github.com/apache/polaris/pull/395/files#diff-38ed2dc48574ff5a3bf83463278cfa4e40ab8bb40abc4fe2aa58b46e6beb927fR793-R800) and [here](https://github.com/apache/polaris/pull/395/files#diff-38ed2dc48574ff5a3bf83463278cfa4e40ab8bb40abc4fe2aa58b46e6beb927fR860-R870). This allows admins to turn off credential vending by default, but to enable it on a case-by-case basis if they trust the external catalog or if the potential for credential overreach is not a security concern.